taskcoordinator/app/src/test/java/com/microservices/taskcoordinator/controllers/CoordinatorControllerIntegrationTests.java

```diff
@@ -77,12 +100,105 @@ public void testCoordinateOrder_allValid_orderCoordinatedAndLaundryStateChanged(
+
+        String orderSubmittedJson = objectMapper.writeValueAsString(orderSubmittedDto);
+        mockMvc.perform(put("/orders/" + RESERVED_ORDER_ID + "/status/submitted")
+                .contentType(APPLICATION_JSON.toString())
+                .content(orderSubmittedJson))
+                .andExpect(status().isOk());
+
+        LaundryStateDto updatedLaundryState = laundryStateService.getLaundryStateById(DEFAULT_LAUNDRY_ID);
+
+        assertEquals(DEFAULT_QUEUE_WAITING_TIME, updatedLaundryState.getQueueWaitingTime());
+        assertEquals(DEFAULT_RESERVED_TIME, updatedLaundryState.getReservedTime());
+        assertEquals(DEFAULT_LAUNDRY_STATE_VERSION, updatedLaundryState.getVersion());
+        assertEquals(OrderStatus.SUBMITTED, orderService.getOrderById(RESERVED_ORDER_ID).getStatus());
+    }
+
+    @Test
+    @Sql(scripts = "/test-data/basic.sql")
+    @Transactional
+    public void testProcessOrder_orderIsSubmitted_orderIsProcessed() throws Exception {
+        InboundLaundryStateDto inboundLaundryStateDto = new InboundLaundryStateDto(DEFAULT_LAUNDRY_ID,
+                DEFAULT_QUEUE_WAITING_TIME, DEFAULT_LAUNDRY_STATE_VERSION);
+        OrderProcessedDto orderSubmittedDto = new OrderProcessedDto(SUBMITTED_ORDER_ID, inboundLaundryStateDto, DEFAULT_COMPLETION_TIME);
+
+        String orderSubmittedJson = objectMapper.writeValueAsString(orderSubmittedDto);
+        mockMvc.perform(put("/orders/" + SUBMITTED_ORDER_ID + "/status/processed")
+                .contentType(APPLICATION_JSON.toString())
+                .content(orderSubmittedJson))
+                .andExpect(status().isOk());
+
+        LaundryStateDto updatedLaundryState = laundryStateService.getLaundryStateById(DEFAULT_LAUNDRY_ID);
+
+        assertEquals(DEFAULT_QUEUE_WAITING_TIME, updatedLaundryState.getQueueWaitingTime());
+        assertEquals(DEFAULT_RESERVED_TIME, updatedLaundryState.getReservedTime());
+        assertEquals(DEFAULT_LAUNDRY_STATE_VERSION, updatedLaundryState.getVersion());
+        assertEquals(OrderStatus.COMPLETE, orderService.getOrderById(SUBMITTED_ORDER_ID).getStatus());
+    }
+
+    @Test
+    @Sql(scripts = "/test-data/one_laundry.sql")
+    @Transactional
+    public void testProcessOrder_orderIsReserved_orderIsNotProcessedDbIsNotChanged() throws Exception {
+        InboundLaundryStateDto inboundLaundryStateDto = new InboundLaundryStateDto(DEFAULT_LAUNDRY_ID, 200L, 2);
+        OrderProcessedDto orderSubmittedDto = new OrderProcessedDto(RESERVED_ORDER_ID, inboundLaundryStateDto, 102L);
+
+        LaundryStateDto initialLaundryState = laundryStateService.getLaundryStateById(DEFAULT_LAUNDRY_ID);
+        OrderDto initialOrder = orderService.getOrderById(RESERVED_ORDER_ID);
+
+        String orderSubmittedJson = objectMapper.writeValueAsString(orderSubmittedDto);
+        try {
+            Exception mvcResult = mockMvc.perform(put("/orders/666/status/processed")
+                    .contentType(APPLICATION_JSON.toString())
```

Comment:
why does 666id didn't change? I guess inconsistent id is being tested (such as in next test)